elide-async/src/main/java/com/yahoo/elide/async/service/DefaultResultStorageEngine.java

```diff
@@ -68,93 +69,117 @@ public URL storeResults(String asyncQueryID, String responseBody) {
+
+        return byteResult;
     }
 
     @Override
-    public AsyncQueryResultStorage deleteResultsByID(String asyncQueryID) {
+    public void deleteResultsByID(String asyncQueryID) {
         log.debug("deleteResultsByID");
 
-        AsyncQueryResultStorage asyncQueryResultStorages = null;
+        AsyncQueryResultStorage asyncQueryResultStorage = null;
 
         try {
-            asyncQueryResultStorages = (AsyncQueryResultStorage) EIT.executeInTransaction(elide,
+            asyncQueryResultStorage = (AsyncQueryResultStorage) DBUtil.executeInTransaction(elide,
                     dataStore, (tx, scope) -> {
 
-                EntityProjection asyncQueryCollection = EntityProjection.builder()
-                        .type(AsyncQueryResultStorage.class)
-                        .build();
+                        EntityProjection asyncQueryCollection = EntityProjection.builder()
+                                .type(AsyncQueryResultStorage.class)
+                                .build();
 
-                Object loaded = tx.loadObject(asyncQueryCollection, asyncQueryID, scope);
-                Object result = loaded;
+                        Object loaded = tx.loadObject(asyncQueryCollection, asyncQueryID, scope);
+                        Object result = loaded;
 
-                if (result != null) {
-                    tx.delete(result, scope);
-                }
+                        if (result != null) {
+                            tx.delete(result, scope);
+                        }
 
-                return loaded;
-            });
+                        return loaded;
+                    });
         } catch (Exception e) {
             log.error("Exception: {}", e);
         }
-        return asyncQueryResultStorages;
     }
 
     @Override
-    public Collection<AsyncQueryResultStorage> deleteAllResults() {
-        log.debug("deleteAllResults");
+    public void deleteResultsCollection(Collection<AsyncQuery> asyncQueryList) {
+        log.debug("deleteResultsCollection");
+
+        Collection<AsyncQueryResultStorage> asyncQueryResultStorage = null;
 
-        Collection<AsyncQueryResultStorage> asyncQueryResultStorages = null;
 
         try {
-            asyncQueryResultStorages = (Collection<AsyncQueryResultStorage>) EIT.executeInTransaction(elide, dataStore,
-                    (tx, scope) -> {
+            asyncQueryResultStorage = (Collection<AsyncQueryResultStorage>) DBUtil.executeInTransaction(elide,
+                    dataStore, (tx, scope) -> {
 
                         EntityProjection asyncQueryCollection = EntityProjection.builder()
                                 .type(AsyncQueryResultStorage.class)
                                 .build();
 
-                        Iterable<Object> loaded = tx.loadObjects(asyncQueryCollection, scope);
-                        Iterator<Object> itr = loaded.iterator();
-
+                        Iterator<AsyncQuery> itr = asyncQueryList.iterator();
```

Comment:
Lets keep this iterator outside executeInTransaction block. we dont want to keep the transaction open for too long. We should iterate, take 1 asyncId, execute in transaction, then move on to next Id.